### ISSUE TITLE
Create a custom media node that allows us to merge embeds

### ISSUE THREAD

**Issue by NiallJoeMaher** [codu-code/codu#158]
## Context

Currently we have custom tags for YouTube, CodeSandbox and CodePen. 
To make it easier it would be great to have a single media tag so that it can either render a specific media it recognises or add it to an iframe. 

## Expected Behavior

Without removing the old tags, add a new tag that allows us all of the functionality of all of the current tags as well as a fallback to an iFrame. 

## Add to the discussion

Either add comments to this issue/and or Discord before taking this one so we can agree on a appropriate interface.

**Comment by xiaoniuniu89** [comment]
I would like to work on this if it is free.

**Comment by NiallJoeMaher** [comment]
Absolutely! I'd love to know if you have suggestions on some ways of it working. You can suggest stuff in open source channel on Discord. 🦾 

I was thinking it would look something like: `{% media src="https://www.youtube.com/embed/93R57sLATM4" /%}`

**Comment by xiaoniuniu89** [comment]
I will spike this and get back to you on some options about how to implement it under the hood but I think the tag as you have it written is perfect.

**Comment by xiaoniuniu89** [comment]
UPDATE: This is the best approach I found so far. Just have to make sure preparing the src urls is nice and tight, refactor comment etc. should finish it in a few days. works with youtube so far. Not sure of what to write for fallback iframe. https://github.com/codu-code/codu/compare/develop...xiaoniuniu89:codu:dcallaghan/MediatagEmbed-%23158

**Comment by NiallJoeMaher** [comment]
@xiaoniuniu89 I might make another issue for the fallback because I'm not sure either to be honest. 

Great work! This is closed with #184 🎉

Some of the examples you suggested in case anyone wants to try it:

```
// Youtube
{% media src="https://www.youtube.com/watch?v=7mmlmxamw_k" /%}

{% media src="www.youtube.com/watch?v=7mmlmxamw_k" /%}

{% media src="https://youtu.be/7mmlmxamw_k" /%}

{% media src="youtu.be/7mmlmxamw_k" /%}

// Code Sandbox
{% media src="https://codesandbox.io/embed/vanilla-vanilla?fontsize=14&hidenavigation=1&theme=dark" / %}

{% media src="https://codesandbox.io/s/vanilla-vanilla" /%}

// Codepen

{% codepen src="https://codepen.io/yuanchuan/pen/dyqYYBN" /%}

{% media src="codepen.io/yuanchuan/pen/dyqYYBN" /%}

// Fallback

{% media src="https://www.codu.co/" /%}
{% media src="www.codu.co/" / %}
```
